### PR TITLE
HARP-13901: Fix THREE.Line and THREE.LineSegments picking.

### PR DESCRIPTION
**Pull request by atomicsulfate**
Raycaster must specify a threshold distance between ray and line.

### CODE REVIEW

File: @here/harp-mapview/lib/MapView.ts [hunk 1]
```diff
@@ -2581,8 +2582,11 @@ export class MapView extends EventDispatcher {
      * points.
      */
     raycasterFromScreenPoint(x: number, y: number): THREE.Raycaster {
```
Comment (FraukeF) [inline @here/harp-mapview/lib/MapView.ts:2584]: I think we should move this whole function to the PickHandler, its only used there, and as its still in @alpha should not be used by anybody

File: @here/harp-mapview/lib/MapView.ts [hunk 2]
```diff
@@ -1087,7 +1088,7 @@ export class MapView extends EventDispatcher {
         // setup camera with initial position
         this.setupCamera();
 
-        this.m_raycaster = new PickingRaycaster(width, height, this.m_env);
+        this.m_pickingRaycaster = new PickingRaycaster(width, height, this.m_env);
```
Comment (FraukeF) [inline @here/harp-mapview/lib/MapView.ts:1091]: move it to the PickingHandler

File: @here/harp-mapview/lib/MapView.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/MapView.ts]: done

File: @here/harp-mapview/lib/MapView.ts (hunk 2 above)
Comment (atomicsulfate) [reply @here/harp-mapview/lib/MapView.ts]: done

File: @here/harp-mapview/lib/PickHandler.ts [hunk 3]
```diff
@@ -217,6 +223,27 @@ export class PickHandler {
         return pickListener.results;
     }
 
+    /**
+     * Returns a ray caster using the supplied screen positions.
+     *
+     * @param x - The X position in css/client coordinates (without applied display ratio).
+     * @param y - The Y position in css/client coordinates (without applied display ratio).
+     *
+     * @alpha
```
Comment (FraukeF) [inline @here/harp-mapview/lib/PickHandler.ts:232]: I think you can make it private

Comment (atomicsulfate) [reply @here/harp-mapview/lib/PickHandler.ts:232]: I can't, TerrainElevationProvider uses that, which I forgot to change btw. Shall I mark it as internal?

Comment (FraukeF) [reply @here/harp-mapview/lib/PickHandler.ts:232]: ahhh, sorry I did only a search in coresdk, my bad.

yes, please mark it as internal then

Comment (atomicsulfate) [reply @here/harp-mapview/lib/PickHandler.ts:232]: actually the whole PickHandler is internal, so no need.

**Review by FraukeF — DISMISSED** [review]

**Comment by nzjony** [comment]
retest this please

**Comment by atomicsulfate** [comment]
@harpgl-bot retest this please

**Comment by atomicsulfate** [comment]
@harpgl-bot retest this please

**Review by FraukeF — APPROVED** [review]

**Pull request merged**